tests/bluetooth/tester/src/l2cap.c

```diff
@@ -141,41 +141,73 @@ static struct channel *get_free_channel()
 	return NULL;
 }
 
+static struct bt_l2cap_chan *get_free_channels(void)
+{
+	uint8_t i;
+	struct channel *chan;
+	struct bt_l2cap_chan *chans[CHANNELS];
```

Comment:
This should be an input argument instead of declared on the stack here. You probably want to fill in the array declared here:
https://github.com/zephyrproject-rtos/zephyr/pull/33979/files#diff-182aa2744c68530a1abe38ed57e096ef585a3c2044512207396e5414ff44dc61R170

And perhaps verify that you can allocate as many free channels as was requested in the command?

Reply:
Thank you for noticing that!

I corrected this behavior, and added check if all requested channels were allocated, but I don't return error if not, just leave this in logs. I made it that way because lesser amount of channels might be sufficient to pass tests (namely 1, because CHANNELS is set to 2)